bubuku/features/restart_if_dead.py

```diff
@@ -18,18 +19,31 @@ def __init__(self, broker: BrokerManager, zk: BukuExhibitor):
     def check(self) -> Change:
         if not self.need_check:
             return None
-        if self.broker.is_running_and_registered():
+        if self.is_running_and_registered(tries=3):
             return None
 
         _LOG.info('Oops! Broker is dead, triggering restart')
         self.need_check = False
 
         # Do not start if broker is running and registered
         def _cancel_if():
-            return self.broker.is_running_and_registered()
+            return self.is_running_and_registered()
 
         return RestartBrokerChange(self.zk, self.broker, _cancel_if, self.on_check_removed)
 
+    def is_running_and_registered(self, tries=1):
+        if not self.broker.process.is_running():
+            return False
+        # Retry with a sleep of 2 seconds in between
+        for x in range(0, tries):
+            if (x > 0):
+                sleep(2)
```

Comment:
Please, do not use `sleep()` function, as it will block execution here (meaning that it will block all the functions of bubuku, including requests from cli interface)

Reply:
just to clarify a bit: the checks are executed with the certain interval by main loop / thread. sleeping here will block every check. the right approach is to count how many times broker was not registered, and if it is some number in a row, like 3 (3 times the check returned false in is_registered) then it should be really the case.

Reply:
I reason that if a broker is not registered into Zookeeper, it makes sense to wait for an interval longer than the configured value of `zookeeper.connection.timeout.ms` until we can run any other change or check. 

The zookeeper client might succeed in restoring the session after a few attempts.

Ideally, we would configure Kafka to die on its own if the Zookeeper client cannot restore the session after a few attempts with an interval that we configured, and this check would only restart the process if it's not running.

PS: If the broker lost the connection with the Zookeeper and the session has died, does it make sense to run other checks or changes?